AutomationCSharpTraining/AutomationCSharpTraining/UnitTest1.cs

```diff
@@ -3,47 +3,103 @@
 using OpenQA.Selenium;
 using OpenQA.Selenium.Chrome;
 using System.Linq;
-
+using System;
+using OpenQA.Selenium.Support.UI;
+using OpenQA.Selenium.Interactions;
 
 namespace AutomationCSharpTraining
 {
     public class Tests
     {
         private IWebDriver driver;
+        private static readonly By fiftyPercentElement = By.XPath("//div[contains(text(),\"50%\")]");
+        private static readonly By downloadButton = By.XPath("//*[@id=\"cricle-btn\"]");  
+        private static readonly By whenFiftyPercentAppearsElement = By.XPath("//*[contains(@class,\"clipauto\")]");//can be used as alternative solution
+        private static readonly By zeroPercentElement = By.XPath("//div[contains(text(),\"0%\")]"); 
 
         [SetUp]
         public void Setup()
         {
+            
             ChromeOptions options = new ChromeOptions();
             options.AddArguments("--incognito");
             driver = new ChromeDriver(options);
+            driver.Manage().Timeouts().ImplicitWait = TimeSpan.FromSeconds(5);
         }
 
         [Test]
         public void SuccessfullLogin()
         {
-            //
-            driver.Url = "https://yandex.by";
+            
             HomePage homePage = new HomePage(driver);
-            Thread.Sleep(2000);            
```

Comment:
Do you really need all these sleeps in your test? I would recommend you to increase value of ImplicitWait. It should be enough. But if you need to wait some element or condition then use explicit waiters.

Reply:
I've deleted all Sleeps before making task 50. Left only one on 36 string with comment according to the task.